discretisedfield/region.py

```diff
@@ -139,6 +164,31 @@ def pmin(self):
         """
         return self._pmin
 
+    @pmin.setter
+    def pmin(self, value):
+        if not isinstance(value, (tuple, list, np.ndarray)):
+            raise TypeError(
+                "pmin must be of type tuple, list, or np.ndarray. Not"
+                f" type(pmin)={type(value)}."
+            )
+        if not all(isinstance(i, numbers.Number) for i in value):
+            raise TypeError("pmin can only contain elements of type numbers.Number.")
+
+        if not len(value) == self.ndim:
+            raise ValueError(
+                "The length of pmin must be the same as the number of"
+                f" dimensions: {self.ndim=}. Not len(pmin)={len(value)}."
+            )
+
+        old_pmin = self._pmin
+        self._pmin = np.array(value)
+        if not np.all(self.edges):
+            edges = self.edges
+            self._pmin = old_pmin
+            raise ValueError(
+                f"At least one of the region's edge lengths is zero: {edges}."
+            )
```

Comment:
Do we really want to allow setting pmin (and pmax)? I would remove it and only allow changes via scaling and translation (which both create new objects).